test/v4/controllers/createUkPropertyPeriodSummary/def2/Def2_CreateUkPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -0,0 +1,530 @@
+  private val parsedUkFhlPropertyMinimal      = parsedUkFhlProperty.copy(income = Some(parsedFhlIncomeMinimal), expenses = None)
+
+  private val parsedNonFhlIncomeRentARoom = Def2_Create_UkPropertyIncomeRentARoom(Some(947.66))
+
+  //@formatter:off
+  private val parsedNonFhlIncome = Def2_Create_UkNonFhlPropertyIncome(
+    Some(42.12), Some(84.31), Some(9884.93),
+    Some(842.99), Some(31.44), Some(parsedNonFhlIncomeRentARoom)
+  )
+
+  private val parsedNonFhlIncomeMinimalDef1 = Def1_Create_UkNonFhlPropertyIncome(
+    None, None, Some(9884.93),
+    None, None, None
+  )
+
+  private val parsedNonFhlIncomeMinimal = Def2_Create_UkNonFhlPropertyIncome(
+    None, None, Some(9884.93),
+    None, None, None
+  )
+  //@formatter:on
+
+  private val parsedNonFhlExpensesRentARoom = Def2_Create_UkPropertyExpensesRentARoom(Some(8842.43))
+
+  //@formatter:off
+  private val parsedNonFhlExpenses = Def2_Create_UkNonFhlPropertyExpenses(
+    Some(3123.21), Some(928.42), Some(842.99), Some(8831.12),
+    Some(484.12), Some(99282), Some(12.34), Some(974.47),
+    Some(12.34), Some(parsedNonFhlExpensesRentARoom), None
+  )
+
+  private val parsedNonFhlExpensesConsolidated = Def2_Create_UkNonFhlPropertyExpenses(
+    None, None, None, None, None, None, None, None, None, Some(Def2_Create_UkPropertyExpensesRentARoom(Some(935.01))), Some(988.18)
+  )
+
+  //@formatter:on
+
+  private val parsedUkNonFhlProperty = Def2_Create_UkNonFhlProperty(
+    Some(parsedNonFhlIncome),
+    Some(parsedNonFhlExpenses)
+  )
+
+  private val parsedUkNonFhlPropertyConsolidated = parsedUkNonFhlProperty.copy(expenses = Some(parsedNonFhlExpensesConsolidated))
+  private val parsedUkNonFhlPropertyMinimal      = parsedUkNonFhlProperty.copy(income = Some(parsedNonFhlIncomeMinimal), expenses = None)
+
+  private val parsedUkNonFhlPropertyMinimalDef1 = Def1_Create_UkNonFhlProperty(income = Some(parsedNonFhlIncomeMinimalDef1), expenses = None)
+
+  private val parsedBodyDef1 =
+    Def1_CreateUkPropertyPeriodSummaryRequestBody("2024-04-06", "2024-07-05", None, Some(parsedUkNonFhlPropertyMinimalDef1))
+
+  private val parsedBody =
+    Def2_CreateUkPropertyPeriodSummaryRequestBody("2024-04-06", "2024-07-05", Some(parsedUkFhlProperty), Some(parsedUkNonFhlProperty))
+
+  private val parsedBodyConsolidated =
+    parsedBody.copy(ukFhlProperty = Some(parsedUkFhlPropertyConsolidated), ukNonFhlProperty = Some(parsedUkNonFhlPropertyConsolidated))
+
+  private val parsedBodyMinimalFhl    = parsedBody.copy(ukFhlProperty = Some(parsedUkFhlPropertyMinimal), ukNonFhlProperty = None)
+  private val parsedBodyMinimalNonFhl = parsedBody.copy(ukFhlProperty = None, ukNonFhlProperty = Some(parsedUkNonFhlPropertyMinimal))
+
+  private val validatorFactory = new CreateUkPropertyPeriodSummaryValidatorFactory(mockAppConfig)
+
+  private def validator(nino: String, taxYear: String, businessId: String, body: JsValue) =
+    validatorFactory.validator(nino, businessId, taxYear, body)
```

Comment:
Can you please change the validator method to use Def2 Validator here please like:

  private def validator(nino: String, businessId: String, taxYear: String, body: JsValue) =
    new Def2_CreateUkPropertyPeriodSummaryRulesValidator(nino, businessId, taxYear, body)(mockAppConfig)

Please make the same change for Def1 Validator